ypathgen/pathgen.go

```diff
@@ -191,21 +196,23 @@ type GoImports struct {
 // specifies the set of paths that are to be searched for associated models (e.g.,
 // modules that are included by the specified set of modules, or submodules of those
 // modules). It extracts the set of modules that are to be generated, and returns
-// a pointer to a GeneratedPathCode struct containing all the generated code to
+// a map of GeneratedPathCode struct containing all the generated code to
 // support the path-creation API. The important components of the generated
 // code are listed below:
 //	1. Struct definitions for each container, list, or leaf schema node,
 //	as well as the fakeroot.
 //	2. Next-level methods for the fakeroot and each non-leaf schema node,
 //	which instantiate and return the next-level structs corresponding to
 //	its child schema nodes.
+// The key of the map is the package name and the value is the information
+// needed to generate code for that package.
```

Comment:
Maybe this should be part of the first sentence so that it's clear from the outset what the key is?

Reply:
agreed